service/index/storage.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package index
+
+import (
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type Storage interface {
+	ReadLibrary
+	WriteLibrary
+}
+
+type ReadLibrary interface {
+	RetrieveFirst(height *uint64) func(*badger.Txn) error
+	RetrieveLast(height *uint64) func(*badger.Txn) error
+	RetrieveHeader(height uint64, header *flow.Header) func(*badger.Txn) error
+	RetrieveCommit(height uint64, commit *flow.StateCommitment) func(*badger.Txn) error
+	RetrieveEvents(height uint64, types []flow.EventType, events *[]flow.Event) func(*badger.Txn) error
+	RetrievePayload(height uint64, path ledger.Path, payload *ledger.Payload) func(*badger.Txn) error
+	RetrieveHeight(blockID flow.Identifier, height *uint64) func(*badger.Txn) error
+	RetrieveTransaction(transactionID flow.Identifier, transaction *flow.Transaction) func(*badger.Txn) error
+	RetrieveTransactions(blockID flow.Identifier, transactions *[]flow.Identifier) func(*badger.Txn) error
+	RetrieveCollection(collectionID flow.Identifier, collection *flow.LightCollection) func(*badger.Txn) error
+	RetrieveCollections(blockID flow.Identifier, collections *[]flow.Identifier) func(*badger.Txn) error
+}
+
+type WriteLibrary interface {
+	SaveFirst(height uint64) func(*badger.Txn) error
+	SaveLast(height uint64) func(*badger.Txn) error
+	SaveCommit(height uint64, commit flow.StateCommitment) func(*badger.Txn) error
```

Comment:
This is all fixed in #228 so I won't change it in this PR if it's OK, unless #228 gets merged before this one in which case I'll have to add those changes in it as well (it should be hard to miss since there will be a conflict 😄)

Reply:
Ah, okay okay, I'm just getting to that PR now.. 

Plus it's really a non-issue. However, I do like to point to things like that just to show that I did in fact read through it and not just blindly approve 😄